### DISCUSSION TITLE
DrawPolygon now fills the polygon, while it used to only stroke the outline, how to get the old behavior?

### DISCUSSION THREAD

**Discussion by woutware** [SixLabors/ImageSharp.Drawing#322] [Q&A] [answered]
Edit: after further investigations this seems to be a bug, I've edited the text below.

When argument thickness is less than 0.5 (e.g. 0.4), method DrawPolygon now fills the polygon, while it used to only stroke the outline, how to get the old behavior? Tested on ImageSharp 3.1.3, ImageSharp.Drawing 2.1.2. When the thickness is equal to 0.5 or greater, the behavior is like it used to be (I think the expected/correct behavior).

I've narrowed it down to the change having happened in ImageSharp.Drawing 2.0.0. It was OK in 1.0.0.

The call that used to not fill the polygon (from DrawPolygonExtensions):

```
 public static IImageProcessingContext DrawPolygon(
     this IImageProcessingContext source,
     Color color,
     float thickness,
     params PointF[] points) =>
     source.DrawPolygon(new SolidBrush(color), thickness, points)
```

**Comment by tocsoft** [comment]
I can't reproduce this.

for me I'm outputting 
![image](https://github.com/SixLabors/ImageSharp.Drawing/assets/166440/74b6ac46-382a-4659-9955-0f466f7d2de6)
for this code
```
 image.Mutate(x => x.DrawPolygon(Color.RebeccaPurple, 2, new PointF[] {
     new(5, 5),
     new(5, 150),
     new(150, 150),
 });
```

Can you provide a example that reproduces this issue?

**Reply by woutware** [reply]
Hi, it seemingly has to do with the value of the (pen) thickness parameter. For values smaller than 0.5 (e.g. take 0.4 or so) it will fill the polygon entirely. I am just heading out of the office now, but I can create a fully functional reproduction later today or tomorrow.

If you want to know why these small thickness values: it basically has to do with zooming. I'm rendering a vector drawing where each line has its own line width, and when you zoom out far enough, line widths become arbitrarily small. I could cap them at 1 pixel minimum as a work around ofcourse, but it seems like a sixlabors drawing bug nonetheless.

**Reply by tocsoft** [reply]
yeah agree its totally a bug and can confirm that this code
```c#
  using var image = new Image<Rgba32>(300, 300, Color.White);
  image.Mutate(x => x.DrawPolygon(Color.RebeccaPurple, 0.4f, new PointF[] {
      new(5, 5),
      new(5, 150),
      new(150, 150),
  }));

  image.Save("path.png");
```
produces 
![image](https://github.com/SixLabors/ImageSharp.Drawing/assets/166440/8592bca9-964a-4a29-a8d1-e976d5189b5c)


We can't promise anything in terms resolving the issue but verifying a reproduction is a great first step

**Comment by tocsoft** [comment, marked as answer]
https://github.com/SixLabors/ImageSharp.Drawing/issues/323 has been created to track this issue.
